src/components/Comment-form.jsx

```diff
@@ -1,11 +1,69 @@
-import React from "react";
+import { useState } from "react";
 
-const CommentForm = () => {
-  return (
-    <form>
-      <label></label>
-    </form>
-  );
+const CommentForm = ({
+  comment,
+  setComments,
+  article_id,
+  postComment,
+  username,
+  setCommentCount,
+}) => {
+  const [newComment, setNewComment] = useState("Enter comment here...");
+  const [isValid, setIsValid] = useState(true);
+  if (comment) {
+    return (
+      <form
+        className="commentForm"
+        onSubmit={(e) => {
+          e.preventDefault();
+          if (newComment === "") {
+            setIsValid(false);
+            setNewComment("Please enter comment");
+          } else if (newComment === "Enter comment here...") {
+            setIsValid(false);
+          } else if (newComment === "Please enter comment") {
+            setIsValid(false);
+          } else if (newComment !== "") {
+            setComments((currComments) => {
+              let comm = {
+                votes: 0,
+                author: username,
+                body: newComment,
+                created_at: "Just now",
+                comment_id: "TBC",
+              };
+              return [comm, ...currComments];
+            });
+            setCommentCount((currentComms) => {
+              return currentComms + 1;
+            });
+            postComment(newComment, username, article_id);
+
+            setNewComment("");
+          }
```

Comment:
We are optimistically rendering the comment but what do you want the UX to be if the comment fails to upload? When do you want to clear the comment in the form? At the moment, if I comment when I'm offline I will lose the form input and have to start again.